astropy/io/misc/hdf5.py

```diff
@@ -178,10 +178,12 @@ def read_table_hdf5(input, path=None):
     else:
         # Read the meta-data from the file
         table.meta.update(input.attrs)
-
+    for column in table.colnames:
+        if hasattr(table[column],"meta"):
+            if table[column].meta.get('__is_encoded_unicode__', False):
+                table.replace_column(column, np.char.decode(table.columns[column], "utf-8"))
```

Comment:
As said [here](https://github.com/astropy/astropy/pull/7024#issuecomment-356004560) `Table` will take care of converting to unicode if needed, so you should not convert back to unicode:
```
In [3]: t = Table.read('test.hdf5', path='the_table')

In [4]: t
Out[4]: 
<Table length=3>
  x   
bytes1
------
     a
     b
     c

In [5]: t['x'] == ['a', 'b', 'c']
Out[5]: array([ True,  True,  True])
```

Reply:
I read the  details of "unicode sandwich" from the links you posted but I thought it is showing dtype of columns as bytes when the user is storing them in form of unicode strings in disk .So i converted data back to unicode strings  in the form user stored it.Should I not convert it back ?

Reply:
Yes, I think we should not convert back, as the same is done for other formats (it allows to minimize the memory usage).

Reply:
fine I'll do it.